heapster-saw/src/Verifier/SAW/Heapster/TypedCrucible.hs

```diff
@@ -2877,14 +2877,16 @@ tcEmitStmt ::
   StmtPermCheckM ext cblocks blocks tops ret RNil RNil (CtxTrans ctx')
 tcEmitStmt ctx loc stmt =
   do _     <- stmtTraceM (const (pretty "Type-checking statement:" <+>
-                        ppStmt (size ctx) stmt))
+                                 ppStmt (size ctx) stmt))
      !_    <- permGetPPInfo
      !pps  <- mapM (\(Some r) -> ppCruRegAndPerms ctx r) (stmtInputRegs stmt)
-     !_    <- stmtTraceM (\_-> pretty "Input perms:" <> softline <> ppCommaSep pps)
+     !_    <- stmtTraceM (\_-> pretty "Input perms:" <> softline <>
```

Comment:
This `stmtTrace` has a `!` but others don't. Should they all be the same? Which one is right?

Reply:
At one point I thought that maybe one of the issues with performance was strictness, so I tried to annotate a bunch of locations with bang patters. This is a holdover from when I did that. It didn't seem to matter, but I also didn't want to mess with anything too much, since it might mess up the order of the trace statements that I have been using to debug all of this code. This is definitely a technical debt issue, that should be addressed at some point, but today is probably not that point...